README.md

```diff
@@ -1,16 +1,295 @@
+'www' urls are purely for view/download purposes. <br />
+Replace 'www' with 'dl' in the event that an expected file format is required for any sort of processing<br />
+Say for example to import json for postman directly using the URL : <br />
+&emsp;Use https://dl.dropbox.com/scl/fi/f7o3rd45sq4zsiuqucnkp/ByteCoders.postman_collection.json?rlkey=gdcqv823snw3oawx6i80rbiyb&st=pjxn6cmh&dl=0 <br />
+&emsp;Instead of https://www.dropbox.com/scl/fi/f7o3rd45sq4zsiuqucnkp/ByteCoders.postman_collection.json?rlkey=gdcqv823snw3oawx6i80rbiyb&st=pjxn6cmh&dl=0 <br />
+
+
+
+## Note About Executing Commands:
+Every command snippet assumes that before running the snippet,<br />
+current working directory is the root directory of this project.
+
+
+# Building and Running a Local Instance
+
+### Install java( preferred: 17/21):
+https://www.oracle.com/java/technologies/downloads/#java21
+
+### Pre-Requisite
+
+Ensure maven is installed <br />
+https://maven.apache.org/install.html
+
+Ensure postgres is installed
+https://www.postgresql.org/download/
+
+Use any db client of your preference <br />
+Here is one for example : <br />
+https://dbeaver.io/
+
+
+### Git Clone Repo :
+```
+https://github.com/NavinColumbia/COMS4156-TeamProject-ByteCoders
+```
+
+### Set-Up:
+
+Ensure to change username and password in application.properties according to database you wish to connect <br />
+We have used the default postgres username <br />
+Ensure that a databse with name 'pharmaid' is already created <br />
+```
+spring.datasource.url=jdbc:postgresql://localhost:5432/pharmaid
+spring.datasource.username=postgres
+spring.datasource.password=postgres
+```
+
+Install Lombok Plugin for Intellij :
+
+https://plugins.jetbrains.com/plugin/6317-lombok
+
+
+Cancel any applications running in port 8080.<br />
+Refer: [killing pid](https://dev.to/osalumense/how-to-kill-a-process-occupying-a-port-on-windows-macos-and-linux-gj8#:~:text=To%20identify%20the%20process%20using,command%20in%20the%20Command%20Prompt.&text=Here%2C%201234%20is%20the%20PID%20of%20the%20process%20using%20port%205672%20.&text=To%20kill%20the%20process%2C%20use,with%20the%20PID%20obtained%20above.&text=Replace%201234%20with%20the%20actual%20PID)
+
+
+### Build:
+```
+mvn clean verify
+```
+
+### Running The Application:
+```
+mvn spring-boot:run
+```
+Application must now be Running In: http://localhost:8080/
+
+
+
+
+
+
+# Style Checking
+
+Run checkstyle:
+```
+mvn checkstyle:check
+```
+Check console output for errors/warnings/violations.
+![current checkstyle reports 0 violation](https://dl.dropbox.com/scl/fi/3xpqsf94i3vqgbj7rti9l/checkstyle_check.png?rlkey=1f60qzif8wviopr23xcl0352y&st=dc2m561y&dl=0)
+
+# Static Code Analysis
+PMD(mentioned in assignment) was the static analyzer used.
+&nbsp;
+
+&emsp;Running PMD:
+
+```
+    pmd check -d src/main/java -R rulesets/java/quickstart.xml -f text -r "pmdres.txt"
+    pmd check -d src/main/java -R rulesets/java/quickstart.xml -f html -r "pmdres.html"
+```
```

Comment:
Let's remove this part about PMD because the rules are checked with `mvn clean verify` and output to `target/pmd.xml`.  We're not checking with quickstart, our rules are already included when we run `mvn clean verify`.

Reply:
alright